components/root/root.jsx

```diff
@@ -334,6 +335,10 @@ export default class Root extends React.Component {
                         path={'/mfa'}
                         component={Mfa}
                     />
+                    <LoggedInRoute
```

Comment:
Right below, we have the `LoggedInRoute` matching on `NeedsTeam`. I'm wondering if that component already does the heavy lifting, and whether or not we could insert the redirector inside there?

Reply:
That was my first stab at this. The issue ended up being that `NeedsTeam` relies on query params to see if it can find a matching team locally. Don't think there is anything substantial redirector can use out of the `NeedsTeam` component, because i will have to figure out the team before i can leverage the team joining logic that's inside `NeedsTeam`.

I think having this component be independent of a team is cleaner where you can go and see how the redirection logic works.

I can however remove redirection for non-DM posts.

2/5 on keeping it as a separate component? Thoughts @lieut-data?

Reply:
Can you expand on the need to leverage "team joining logic"? Perhaps I've misunderstood the domain, but I was hoping we can just rely on the teams of which the user is already a part. If they don't belong to any teams, it seems reasonable not to resolve a permalink at all in that case.

Reply:
Ah i see. That's not how permalinks are currently working. If you're not part of a team, permalink resolution results in you joining a team, which i also agree is a bit odd.

by "team joining logic" i meant it fetches `TeamUnreads`, and performs other actions like `SelectTeam`, `setPreviousTeamId`.

It doesn't load anything from the web that `Redirector` needs, which is why i'm leaning towards not integrating `Redirector` as part of `NeedsTeam` as we're not duplicating logic.